webpack/stories/docs/APIMiddlewareUsage.md

```diff
@@ -0,0 +1,146 @@
+# Example: how to use the API middleware
+
+The example API returns a JSON object like this:
+```json
+{
+  "items": [
+    {"id": 319, "title": "setting", "action": "update"},
+    {"id": 150, "title": "bookmark", "action": "create"}
+  ],
+}
+```
+
+```js
+  /** MyComponent.js*/
+  import React from 'react';
+  import { STATUS } from '../../constants';
+
+  const MyComponent = ({ status, error, items }) => {
+    if (status === STATUS.PENDING ) {
+      return <div>Loading...</div>
+    }
+
+    if (status === STATUS.ERROR) {
+      return <div>Error: {error.message}</div>
+    }
+
+    return (
+      <ul>
+        {items.map(item => (
+          <li key={item.id}>
+            {item.title} {item.action}
+          </li>
+        ))}
+      </ul>
+    );
+  }
+```
+
+```js
+  /** index.js*/
+  import React, { useEffect } from 'react';
+  import PropTypes from 'prop-types';
+  import { useSelector, useDispatch } from 'react-redux';
+  import { selectItems, selectStatus, selectError } from './MyComponentsSelectors.js';
+  import { getData } from './MyComponentActions';
+  import MyComponent from './MyComponent';
+
+  const ConnectedMyComponent = ({ path }) => {
+    const items = useSelector(selectItems);
+    const status = useSelector(selectStatus);
+    const error = useSelector(selectError);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+      dispatch(getData(path))
+    });
```

Comment:
Shouldn't it suppose to be:
```js
useEffect(() => {
  dispatch(getData(path))
}, [path]);
```

Reply:
Also the `getData` should be in the array?

Reply:
according to the react docs:

Functions passed to `useEffect` are executed on every component rendering—unless you pass a second argument to it.

If the second argument is an empty array, like the example above, it will behave exactly like the `componentDidMount`, only executing on the first rendering.

Alternatively, you can pass one or more values inside the array, which will make `useEffect` execute every time those value changes.

In our case I believe we want to trigger it only once as in `componentDidMount` so i will update it with an empty array.

Reply:
So doesn't it make sense to fetch data on each `path` change? each `path` refect different data.

IMO the most common example is something like:
```jsx
<HostPage hostId='some-id' />
```

I would want to fetch the data again if `react-router` will change the `hostId`. (breadcrumb switcher?)

**IMO all the props used by the `useEffect` should be listed in the array.**
We have an `eslint` rule that should warn you if you don't do it.
See: https://github.com/theforeman/foreman/pull/7225